exasol_sphinx_github_pages_generator/deployer.py

```diff
@@ -0,0 +1,192 @@
+                      f"and checked out (Local or stashed changes will be ignored in this build).")
+                os.chdir(f"{self.worktree_paths['source_worktree']}/doc")
+                current_branch = run(["git", "rev-parse", "--abbrev-ref", "HEAD"], capture_output=True, text=True,
+                                     check=True)
+                print("new current_branch/detected source branch : " + str(current_branch.stdout))
+            except subprocess.CalledProcessError:
+                sys.exit("Problem with adding worktree for source")
+
+        elif source_branch_commit_id.returncode == 0:
+            sys.exit("Source branch exists locally, but not on remote, and source branch is not current branch."
+                     "Please push your source branch to remote.")
+        else:
+            sys.exit(f"source branch {self.source_branch} does not exist")
+
+    def detect_or_verify_source_branch(self):
+        current_branch = run(["git", "rev-parse", "--abbrev-ref", "HEAD"], capture_output=True, text=True, check=True)
+        print(current_branch.stdout)
+        print(f"source branch before detect: {self.source_branch}")
+        if self.source_branch == "":
+            if current_branch.stdout == "":
+                sys.exit("Abort. Could not detect current branch and no source branch given.")
+            print(f"No source branch given. Using Current branch {current_branch.stdout[:-1]} as source.")
+            self.source_branch = current_branch.stdout[:-1]
+
+        source_branch_commit_id = run(["git", "rev-parse", self.source_branch], capture_output=True, text=True)
+        # the [:-1] removes the newline from the output
+        if current_branch.stdout[:-1] != self.source_branch:
+            print("Current branch is not source branch. Need to switch branches")
+            self.check_out_source_branch_as_worktree(source_branch_commit_id)
+            return
+        if source_branch_commit_id.stdout != self.current_commit_id:
+            sys.exit(f"Abort. Current commit id {self.current_commit_id} and commit id of source "
+                     f"branch {source_branch_commit_id} are not equal. Please commit your changes.")
+        print(f"Detected source branch {self.source_branch}")
+
+    def checkout_target_branch_as_worktree(self):
+        target_branch_exists = run(["git", "show-branch", f"remotes/origin/{self.target_branch}"], capture_output=True,
+                                   text=True)
+        print("target_branch_exists : " + target_branch_exists.stdout + str(
+            target_branch_exists.returncode) + target_branch_exists.stderr)
+        if target_branch_exists.returncode == 0:
+            print(f"Create worktree from existing branch {self.target_branch}")
+            run(["git", "worktree", "add", self.worktree_paths["target_worktree"], self.target_branch], check=True)
+        else:
+            print(f"Create worktree from new branch {self.target_branch}")
```

Comment:
When a CLI writes output it always should consider if its `output data` or `status information`.

`output data` -> `stdout`
`status information` -> `stderr`

I know this may sound slightly counter intuitive, because "status" is not == `error` but that's how unix tools do it already for decades. If you break with this rule it is very hard/error prone to hook up the tool into a shell pipe processing chain.
(I am happy to elaborate or give examples on this if you want / need them)